addons/zylann.hterrain/tools/brush/brush_editor_overlay.gd

```diff
@@ -0,0 +1,87 @@
+@tool
+extends Control
+
+const MIN_UI_CIRCLE_SIZE = 40
+const MAX_UI_CIRCLE_SIZE = 700
+
+const MIN_SIZE_VALUE = 2
+const MAX_SIZE_VALUE = 300
+const MIN_OPACITY_VALUE = 0
+const MAX_OPACITY_VALUE = 100
```

Comment:
Not sure if this is the right place to put these constants, they are not used inside this script. They should rather be in the brush script. Size is already a constant there. Opacity isn't because it's technically a ratio, in 0..1. It's only shown as 0..100 in the UI for convenience.

> But I can't seen to find where it gets the min_value of 2 from.

That's a technical limitation of the plugin, but generally the rest still applies, if it doesn't have a constant as the single source of truth, then it should be created.

> But 500 is a bit too big for the overlay as it is now.

How so? For sure if you take that value directly as the size of the overlay then yes it will be quite large, but the overlay's size should only be proportional to it and remap to the same visual range of sizes.  Also, the existing brush size slider in the bottom panel is exponential, which creates a more useful distribution of values when adjusting the UI. Maybe that would work too in the overlay?

Reply:
At some point, I was using a `remap` to make the ui circles work with any value while keeping the same size. 
But there was some weird precision errors happening, like getting the min_size at 1.56 instead of 2 and other strange stuff. Math is not really my thing. 

The other thing, is that it was kinda harder to set values at the lower range.
If I made the circle ui size big, the opacity circle that only goes to 100 feels weird.
If I made the circle ui normal size, then each pixel could increase the brush size to few points at a time.
So I reverted back to the way it was that was quite decent overall.

The exponential ratio you showed in range.cpp feels like it could solve the above problems.
But I'm still worried because last time I tried to use remap, something was not quite right.

I'll see if I can give it another try, but if I cant get it to work I might keep it as it is for now if that's okay.

----
About the constants.

Yeah you are right. I'll set those constants in the brush class.